plugins/services/src/js/utils/ServiceStatusIconUtil.ts

```diff
@@ -0,0 +1,59 @@
+import DateUtil from "#SRC/js/utils/DateUtil";
+
+import Pod from "../structs/Pod";
+import Service from "../structs/Service";
+import ServiceTree from "../structs/ServiceTree";
+import DeclinedOffersUtil from "../utils/DeclinedOffersUtil";
+
+interface ServiceQueue {
+  since: string | null;
+}
+
+const isUnableToLaunch = (service: Service | Pod | ServiceTree) => {
+  const queue = service.getQueue() as ServiceQueue | null;
+  const UNABLE_TO_LAUNCH_TIMEOUT = 1000 * 60 * 30;
+
+  if (queue == null) {
+    return false;
+  }
+
+  return (
+    Date.now() - (DateUtil.strToMs(queue.since) || 0) >=
+    UNABLE_TO_LAUNCH_TIMEOUT
+  );
+};
+
+export function getStatusIconProps(service: Service | ServiceTree | Pod) {
+  const queue = service.getQueue() as ServiceQueue | null;
+  const isServiceTree = service instanceof ServiceTree;
+  const unableToLaunch = isUnableToLaunch(service);
+  const appsWithWarnings = isServiceTree
+    ? (service as ServiceTree)
+        .filterItems((item: any) => {
```

Comment:
our typings say that this is a `Service`. why override it with `any`?

possibly the types of `ServiceTree.filterItems` are wrong? (i updated them and possibly forgot that a `serviceTree` can also contain another `ServiceTree`. if so, please let's fix the types instead of working around here.